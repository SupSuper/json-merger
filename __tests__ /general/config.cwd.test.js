const jsonMerger = require("../../dist");
const {testConfig} = require("../__helpers__");

jest.mock("fs");
const fs = require("fs");

describe("when config.cwd is set", function () {

    test("it should try to load files relative to config.cwd", function () {

        const files = {
            "test/a.json": {
                "object": "this should be the value of /object"
            }
        };

        const object = {
            "$import": "a.json"
        };

        fs.__setJsonMockFiles(files);

        const result = jsonMerger.mergeObject(object, testConfig({
            cwd: "./test"
        }));

        expect(result).toMatchSnapshot();
    });
});
