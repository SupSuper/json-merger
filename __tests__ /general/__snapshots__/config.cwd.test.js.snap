// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when config.cwd is set it should try to load files relative to config.cwd 1`] = `
"{
	\\"object\\": \\"this should be the value of /object\\"
}"
`;
